client/components/domains/register-domain-step/index.jsx

```diff
@@ -385,6 +385,28 @@ const RegisterDomainStep = React.createClass( {
 					return suggestion.domain_name;
 				} );
 
+				if ( abtest( 'domainSuggestionNudgeLabels' ) === 'updated' ) {
+					const matchesSearchedDomain = ( suggestion ) => ( suggestion.domain_name === domain ),
+						exactMatchBeforeTld = ( suggestion ) => ( startsWith( suggestion.domain_name, `${ domain.replace( / /g, '') }.` ) ),
+						bestAlternative = ( suggestion ) => ( ! exactMatchBeforeTld( suggestion ) && suggestion.domain_name !== domain );
+
+					const availableDomain = find( suggestions, matchesSearchedDomain );
```

Comment:
You can simplify the logic here a bit - no need for a separate check for the whole domain vs exact match before TLD. You just need to find the first suggestion that has an exact match before the TLD. That's because the `result` array is guaranteed to be in this order:
  * `[0]` - non-falsey value containing a suggestion with an exact match to the query (this is coming from the availability check) - or `null` if it's not a valid domain, it's not mappable, etc.
  * `[1]` - an array of other suggestions.

On line 384, `result` is flattened, etc. but the order remains. So you are sure that if you find an exact match before the TLD, it's either going to be the exact match for the whole thing (if it's in the list of suggestions) or the next best thing.
So (unless I'm missing something), you just need:
```
const recommendedSuggestion = find( suggestions, exactMatchBeforeTld );
if ( recommendedSuggestion ) {
	recommendedSuggestion.isRecommended = true;
}
const bestAlternativeSuggestion = find( suggestions, bestAlternative );
if ( bestAlternativeSuggestion ) {
	bestAlternativeSuggestion.isBestAlternative = true;
}
```